vector/src/main/java/im/vector/riotx/features/roomprofile/AvatarSelectorView.kt

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright (c) 2020 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.riotx.features.roomprofile
+
+import android.animation.Animator
+import android.animation.AnimatorListenerAdapter
+import android.annotation.TargetApi
+import android.content.Context
+import android.graphics.drawable.BitmapDrawable
+import android.os.Build
+import android.util.Pair
+import android.view.Gravity
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewAnimationUtils
+import android.view.animation.Animation
+import android.view.animation.AnimationSet
+import android.view.animation.OvershootInterpolator
+import android.view.animation.ScaleAnimation
+import android.view.animation.TranslateAnimation
+import android.widget.FrameLayout
+import android.widget.ImageButton
+import android.widget.LinearLayout
+import android.widget.PopupWindow
+import androidx.core.view.doOnNextLayout
+import com.amulyakhare.textdrawable.TextDrawable
+import com.amulyakhare.textdrawable.util.ColorGenerator
+import im.vector.riotx.R
+import im.vector.riotx.core.extensions.getMeasurements
+import im.vector.riotx.core.utils.PERMISSIONS_FOR_TAKING_PHOTO
+import im.vector.riotx.core.utils.PERMISSIONS_FOR_WRITING_FILES
+import im.vector.riotx.features.roomprofile.AvatarSelectorView.Callback
+import kotlin.math.max
+
+private const val ANIMATION_DURATION = 250
+
+/**
+ * This class is the view presenting choices for picking avatar.
+ * It will return result through [Callback].
+ */
+class AvatarSelectorView(context: Context,
```

Comment:
Why creating again this class, please refact the existing one to be more dynamic, or use a simple dialog?